sqs/src/main/scala/akka/stream/alpakka/sqs/ControlledThrottling.scala

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright (C) 2016-2019 Lightbend Inc. <http://www.lightbend.com>
+ */
+
+package akka.stream.alpakka.sqs
+
+import java.time.Clock
+
+import akka.stream.Attributes
+import akka.stream.impl.fusing.GraphStages.SimpleLinearGraphStage
+import akka.stream.stage.{GraphStageLogic, InHandler, OutHandler, TimerGraphStageLogic}
+
+import scala.concurrent.duration.FiniteDuration
+
+class ControlledThrottling[T](clock: Clock, delay: FiniteDuration) extends SimpleLinearGraphStage[T] {
+  private var isThrottling: Boolean = false
```

Comment:
This is mutable state which would be shared by all streams using this stage. You are not allowed to have any mutable state here. We need to find a different implementation to achieve this feedback loop. I guess reaching out to the graph DSL is appropriate.

Reply:
I am not sure I understand why this is going to be reused, the idea I had was that anyone who uses this class spawns a new instance of `ControlledThrottling` and therefore the state is not shared

Reply:
This problem is the difference between creating the class and materialising the stream. You may create the source and even a whole stream and materialise it multiple times.

Reply:
You're right, I've created a custom stage (which is basically autobalancing `MapAsyncUnordered`).